src/contradictory_claims/data/preprocess_cord.py

```diff
@@ -199,3 +204,51 @@ def clean_text(input_data):
     input_processed = input_data.loc[sentences_to_keep, :]
 
     return input_processed
+
+
+def merge_section_text(input_data: pd.DataFrame):
+    """
+    Merge all sentences belonging to each section of each paper into contiguous text passages.
+
+    :param input_data: pandas dataframe with publication text
+    :return: Dataframe with merged section text
+    """
+    # Merge all sentences belonging to each section of each paper into contiguous text passages
+    merged_df = input_data.groupby(['cord_uid', 'section'], as_index=False).agg({'sentence': ' '.join})
+    # Convert text to lower case
+    merged_df.loc[:, 'sentence'] = merged_df.loc[:, 'sentence'].str.lower()
+    # Rename column
+    merged_df = merged_df.rename(columns={'sentence': 'text'})
+
+    return merged_df
+
+
+def filter_section_with_drugs(input_data: pd.DataFrame, drug_terms: List[str], drug_terms_pattern: str):
+    """
+    Filter to sections where section text contains drug terms.
+
+    :param input_data: pandas dataframe with publication text
+    :param drug_terms: list of drug terms to search for
+    :param drug_terms_pattern: drug terms regex pattern to search for
+    :return: Dataframe with sections containing drug terms
+    """
+    # Replace drug name short forms with full forms
+    drug_replace_dict = {'hcq': 'hydroxychloroquine', ' cq ': 'chloroquine', ' azt ': 'azithromycin',
+                         ' azi ': 'azithromycin', ' az ': 'azithromycin'}
```

Comment:
Yeah, agree it makes sense to keep it in a file. For 'hcq', it shouldn't matter because there really aren't any words that contain hcq in them. Flanking with white spaces will not detect cases with a punctuation next to them, like "hcq-therapy" or "hcq,".  With the other drug terms, not flanking'll cause issues because they're commonly used in many words.